ext/liquid_c/vm.c

```diff
@@ -67,25 +50,29 @@ static VALUE vm_internal_new(VALUE context)
     VALUE obj = TypedData_Make_Struct(cLiquidCVM, vm_t, &vm_data_type, vm);
     vm->stack = c_buffer_init();
 
-    vm->strainer = rb_funcall(context, id_strainer, 0);
-    Check_Type(vm->strainer, T_OBJECT);
+    vm->context.strainer = rb_funcall(context, id_strainer, 0);
+    Check_Type(vm->context.strainer, T_OBJECT);
 
-    vm->filter_methods = rb_funcall(RBASIC_CLASS(vm->strainer), id_filter_methods_hash, 0);
-    Check_Type(vm->filter_methods, T_HASH);
+    vm->context.filter_methods = rb_funcall(RBASIC_CLASS(vm->context.strainer), id_filter_methods_hash, 0);
+    Check_Type(vm->context.filter_methods, T_HASH);
 
-    vm->interrupts = rb_ivar_get(context, id_ivar_interrupts);
-    Check_Type(vm->interrupts, T_ARRAY);
+    vm->context.interrupts = rb_ivar_get(context, id_ivar_interrupts);
+    Check_Type(vm->context.interrupts, T_ARRAY);
 
-    vm->resource_limits_obj = rb_ivar_get(context, id_ivar_resource_limits);;
-    ResourceLimits_Get_Struct(vm->resource_limits_obj, vm->resource_limits);
+    vm->context.resource_limits_obj = rb_ivar_get(context, id_ivar_resource_limits);;
+    ResourceLimits_Get_Struct(vm->context.resource_limits_obj, vm->context.resource_limits);
 
-    vm->strict_filters = RTEST(rb_funcall(context, id_strict_filters, 0));
-    vm->global_filter = rb_funcall(context, id_global_filter, 0);
+    vm->context.strict_variables = false;
+    vm->context.strict_filters = RTEST(rb_funcall(context, id_strict_filters, 0));
+    vm->context.global_filter = rb_funcall(context, id_global_filter, 0);
```

Comment:
This initialization could be moved into context_internal_init to keep the code coupled to Liquid::Context together.